### ISSUE TITLE
Degraded developer experience / HMR consistently causing Log In screen to show.

### ISSUE THREAD

**Issue by nboliver-ventureweb** [Shopify/shopify-app-template-remix#886]
Every change to a component is causing the log in screen to show. After a hard refresh, things work as expected until the component is saved again. The component is a form that uses forwardRef and is shared by multiple routes.  Changing code in the route files does not cause the log in screen to show. 

![Image](https://github.com/user-attachments/assets/ee97e919-db3c-4812-9af0-f60c2f9990ce)

**Comment by nboliver-ventureweb** [comment]
Today I am frequently getting a Cloudflare tunnel error, which can be corrected with a hard refresh. Between that and the Log in screen, development is pretty slow and the Dx is not great. I am enjoying the app stack, but it seems like there's a few things to iron out to make it effective. 

![Image](https://github.com/user-attachments/assets/9b27e4cd-a5f9-450e-a557-357112cc350c)

**Comment by nboliver-ventureweb** [comment]
Any update here? Development is painfully slow when every save results in a login screen, which then requires a full page refresh...

**Comment by nboliver-ventureweb** [comment]
It looks like there are two separate issues causing the login screen to show. 
One is this: https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react#consistent-components-exports. I had a Zod schema that is located in a component file that was being unnecessarily exported. Perhaps it could be beneficial to add the linked eslint plugin? https://github.com/ArnaudBarre/eslint-plugin-react-refresh

The other thing seems to pop up randomly - the logs show that `shop` is missing:

```
10:01:14 │      remix │ [shopify-app/DEBUG] Missing or invalid shop, redirecting to login path |
{shop: null}
10:01:14 │      remix │ [shopify-app/DEBUG] Authenticate returned a response
10:01:14 │      remix │ [shopify-app/DEBUG] Missing or invalid shop, redirecting to login path |
{shop: null}
10:01:14 │      remix │ [shopify-app/DEBUG] Authenticate returned a response
```

**Comment by nboliver-ventureweb** [comment]
@vincaslt yeah if you look lower down the docs page for the plugin, there is an example of using `allowExportNames` with Remix: https://github.com/ArnaudBarre/eslint-plugin-react-refresh?tab=readme-ov-file#allowexportnames-v044 
In my config I also added `handle` to the list of allowed names.

**Comment by lizkenyon** [comment]
Hi there 👋 

Thank you for your patience on this!

I was able to reproduce this, it is seems like the redirection to the login screen happens when vite does a full `page reload` and triggers a document request. Because the document request is not initiated from the Shopify admin, it is missing the required `shop` parameter which is why you are then getting kicked over to the login screen.

We are going to continue to try to determine the best solution here. Thanks!

**Comment by vinhphon** [comment]
Hi @lizkenyon 

Any update on the issue?

**Comment by Kashuab** [comment]
> [@Kashuab](https://github.com/Kashuab) I think you are supposed to use `Link` (either from remix or polaris) instead of `a`.
> 
> Check this section of the readme: ![Image](https://github.com/user-attachments/assets/9e2a8451-0ecb-4714-9719-9ff34c15ddbd)

@JannikWempe I think we're both wrong haha. The `<TitleBar>` I'm rendering is from App Bridge. According to their docs you're only allows to render `<button>` elements, even for breadcrumbs, which seems a bit silly.

**Comment by lizkenyon** [comment]
Hi folks 👋 

After investigation with the Remix team, we believe the resolution is **upgrading to Vite 6**.

If you update your app to the latest versions of vite and the remix packages, this issue should be resolved.

I am going to close this issue. Please create a new issue with steps for reproduction if you continue to experience this after updating.
